client/src/main/scala/sbtcompendium/client/CompendiumClient.scala

```diff
@@ -50,7 +52,7 @@ trait CompendiumClient[F[_]] {
    * @param identifier the protocol identifier
    * @return a client for that protocol and target
    */
-  def generateClient(target: IdlName, identifier: String, v: Option[String]): F[List[String]]
+  def generateClient(protoConfig: ProtoConfig)(target: IdlName, identifier: String, v: Option[String]): F[List[String]]
```

Comment:
This method should be agnostic of the protocol (protobuff, avro, etc.), right?

Reply:
Since other schema types will have its own configuration, does it look good to you?
`trait SchemaConfig`
`case class ProtoConfig (...) extends SchemaConfig`
`case class OpenApiConfig (...) extends SchemaConfig`
so `generateClient` could be like
`def generateClient(target: IdlName, identifier: String, v: Option[String], schConfig: Option[SchemaConfig] = None): F[List[String]]`

Reply:
LGTM

Reply:
Must it be optional? Should I set a single sbt setting for the schema configuration?

Reply:
right now, that makes sense for me